### PR TITLE
[TASK] Revise Mail API chapter

### PR DESCRIPTION
**Pull request by brotkrueml**
Related: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/104

Releases: main

### CODE REVIEW

File: Documentation/ApiOverview/Mail/Index.rst [hunk 1]
```diff
@@ -26,55 +15,58 @@ for creating email messages.
...
+..  _mail-configuration-format:
 
 Format
 ------
 
-:php:`$GLOBALS['TYPO3_CONF_VARS']['MAIL']['format']` can be `both`,
-`plain` or `html`. This option can be overridden by Extension authors
-in their use cases.
+:php:`$GLOBALS['TYPO3_CONF_VARS']['MAIL']['format']` can be `both`, `plain` or
+`html`. This option can be overridden by extension authors in their use cases.
```
Comment (linawolf) [inline Documentation/ApiOverview/Mail/Index.rst:43]: Should it really be overridden in an Extension and not rather in the LocalConfiguration / Additional Configuration by administrators?

File: Documentation/ApiOverview/Mail/Index.rst [hunk 2]
```diff
@@ -83,118 +75,119 @@ The most important configuration option for sending emails is
...
+    Custom class which implements
+    :php:`\Symfony\Component\Mailer\Transport\TransportInterface`.
+    The constructor receives all settings from the ``MAIL`` section to make it
+    possible to add custom settings.
 
 
-.. _mail-validators:
+..  _mail-validators:
 
 validators
```
Comment (linawolf) [inline Documentation/ApiOverview/Mail/Index.rst:187]:
```suggestion
Validators
```

File: Documentation/ApiOverview/Mail/Index.rst [hunk 3]
```diff
@@ -205,324 +198,352 @@ validators are available:
...
 
-   // Create the message
-   $mail = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(\TYPO3\CMS\Core\Mail\MailMessage::class);
+..  code-block:: php
+    :caption: EXT:site_package/Classes/Utility/MyMailUtility.php
 
-   // Prepare and send the message
-   $mail
+    // Create the message
+    $mail = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(\TYPO3\CMS\Core\Mail\MailMessage::class);
```
Comment (linawolf) [inline Documentation/ApiOverview/Mail/Index.rst:376]: Let's use "Use" statements, that greatly improves the  readability

File: Documentation/ApiOverview/Mail/Index.rst [hunk 4]
```diff
@@ -205,324 +198,352 @@ validators are available:
...
-      // And finally send it
-      ->send()
+        // And finally send it
+        ->send()
     ;
 
 
 
-Or if you prefer, don't concatenate the calls:
+Or, if you prefer, don't concatenate the calls:
```
Comment (linawolf) [inline Documentation/ApiOverview/Mail/Index.rst:408]:
```suggestion
Or, if you prefer, do not concatenate the calls:
```

File: Documentation/ApiOverview/Mail/Index.rst [hunk 5]
```diff
@@ -205,324 +198,352 @@ validators are available:
...
+    Mail; Default sender
+    TYPO3_CONF_VARS; MAIL defaultMailFromAddress
+..  _mail-sender:
 
 How to set and use a default sender
 ===================================
 
-It is possible to define a default email sender ("From:") in the *Install
-Tool*:
+It is possible to define a default email sender ("From:") in the *Install Tool*:
```
Comment (linawolf) [inline Documentation/ApiOverview/Mail/Index.rst:490]:
```suggestion
It is possible to define a default email sender ("From:") in 
:guilabel:`Admin Tools > Settings > Configure Installation-Wide Options`:
```

File: Documentation/ApiOverview/Mail/Index.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/Mail/Index.rst:43]: Good catch.

**Review by linawolf — APPROVED** [review]

**Pull request merged**